go/pkg/symlinkopts/opts.go

```diff
@@ -0,0 +1,119 @@
+// Package symlinkopts provides an efficient interface to create unambiguous symlink options.
+package symlinkopts
+
+import (
+	"fmt"
+	"strings"
+)
+
+// Options represents a set of options for handling symlinks.
+// To ensure a valid set of options, use one of the functions provided in this package.
+type Options uint32
+
+const (
+	Preserve Options = 1 << (32 - 1 - iota)
```

Comment:
why not 1 << iota?

Reply:
It's MSB vs LSB. No particular reason to choosing one over the other. The `fs` package uses MSB for `FileInfo`, but that's so the lower bits can match POSIX file permissions: https://cs.opensource.google/go/go/+/refs/tags/go1.20.4:src/io/fs/fs.go;l=175;drc=aedf298daf508b564e4dddc7687fff8822315a5e